### DISCUSSION TITLE
Processor without npm module

### DISCUSSION THREAD

**Discussion by Splines** [eslint/eslint#17774] [Q&A] [answered]
Hey there, I was wondering if it was possible to define a preprocessor as JS object and include it directly in the ESLint config. In the [Custom Processor documentation](https://eslint.org/docs/latest/extend/custom-processors) it is assumed that one has to create a separate npm module for that, which feels like a lot of extra work if one just wants to get a very small and basic processor to work.

E.g. I'm looking to do something along the lines of this:

```js
// inside .eslintrc.js
const myCustomProcessor = { /* processor methods */ };
module.exports = {
  parserOptions: {...},
  // other stuff,
  overrides: [
    {
      files: ["**/*.my.file.extension"],
      processor: myCustomProcessor // not a string, but instead the Object defined above (or maybe in another file)
    }
  ],
};
```

**Comment by nzakas** [comment, marked as answer]
In eslintrc, the processor must be part of a plugin. There's no way around it.

However, if you use the [new config system](https://eslint.org/docs/latest/use/configure/configuration-files-new#using-processors), you can just provide an object as the value of `processor`.

**Reply by Splines** [reply]
Ah thanks, good to know. In the end, the [processor](https://github.com/Splines/eslint-plugin-erb) turned out to be big enough to be outsourced to its own npm module, so I didn't have the problem anymore ;)
